core/src/sumeragi/message.rs

```diff
@@ -143,14 +135,12 @@ impl From<CommittedBlock> for BlockCommitted {
 #[non_exhaustive]
 pub struct BlockSyncUpdate {
     /// The corresponding block.
-    pub block: VersionedCandidateCommittedBlock,
+    pub block: VersionedCommittedBlock,
```

Comment:
If we anyway do full re-validation on block sync there is no point in transferring  `VersionnedCommittedBlock` we can just use `VersionedPendingBlock`.

Reply:
Bad idea. The revalidation could be a precaution, but semantically  the return type is `VersionedCommittedBlock`. Keeping it that way helps with formal verification.

Reply:
Yes, untyped, unverified and untrusted and thats why we want to indicate that by type and enforce revalidation When receiving such type through the network.

Reply:
In my opinion this would cause again the large amount of types. I think we should avoid turning our programming problems into typesystem problems. But if this was introduced again by another PR I would not be against it.

Reply:
please do not resolve discussions without getting a green light from the OP. 

Also, in this case, it is @Erigara 's opinion as well as mine that you should return the candidate blocks in this PR. Nobody is going to be reverting your changes after you. I will not be going into discussion why it's better to enforce invariants in the type system, it's a common sense. All you should know atm is that majority vote is against you on this topic and we govern ourselves democratically as a team

Reply:
We wouldn't have made it a hard requirement to resolve all conversations before merging if it wasn't necessary. 

Also, the **whole point of code review** is to identify areas where a separate pair of eyes has a different opinion. @Erigara and @mversic have made it clear that they feel strongly about one way of doing things. You don't get to ignore that opinion, you **muat address it**.